src/main/java/org/openbmp/db_rest/resources/LinkState.java

```diff
@@ -145,132 +145,162 @@ else if (peerHashId != null)
                 DbUtils.select_DbToJson(mysql_ds, query));
     }
 
+    @GET
+    @Path("/nodes/virtual_nodes/peer/{peerHashId}/{mt_id}")
+    @Produces("application/json")
+    public Response getLsNodesByPeer(@PathParam("peerHashId") String peerHashId,
+                                     @PathParam("mt_id") Long mt_id) {
+
+        String where = "";
+
+        if (peerHashId != null)
+            where = " and `ls_nodes`.`peer_hash_id` = '" + peerHashId + "' AND `links`.`mt_id` = " + mt_id;
+
+        String query = "select `r`.`name` AS `RouterName`,`r`.`ip_address` AS `RouterIP`,`p`.`name` AS `PeerName`,`p`.`peer_addr` " +
```

Comment:
This needs to be cleaned up...   It appears you are using the view query with a modification.  Not all fields should be required as this was supposed to be just for connecting links in the UI.  Rework this to be only what is needed.